geode-core/src/main/java/org/apache/geode/distributed/internal/InternalConfigurationPersistenceService.java

```diff
@@ -543,6 +544,65 @@ void initSharedConfiguration(boolean loadSharedConfigFromDir)
     this.status.set(SharedConfigurationStatus.RUNNING);
   }
 
+  void removeInvalidXmlConfigurations(Region<String, Configuration> configRegion)
+      throws IOException, SAXException, ParserConfigurationException, TransformerException {
+    for (Map.Entry<String, Configuration> entry : configRegion.entrySet()) {
+      String group = entry.getKey();
+      Configuration configuration = entry.getValue();
+      String configurationXml = configuration.getCacheXmlContent();
+      if (configurationXml != null && !configurationXml.isEmpty()) {
+        Document document = XmlUtils.createDocumentFromXml(configurationXml);
+        boolean removedInvalidReceivers = removeInvalidGatewayReceivers(document);
+        boolean removedDuplicateReceivers = removeDuplicateGatewayReceivers(document);
+        if (removedInvalidReceivers || removedDuplicateReceivers) {
+          configuration.setCacheXmlContent(XmlUtils.prettyXml(document));
+          configRegion.put(group, configuration);
+        }
+      }
+    }
+  }
+
+  boolean removeInvalidGatewayReceivers(Document document) throws TransformerException {
+    boolean modified = false;
+    NodeList receiverNodes = document.getElementsByTagName("gateway-receiver");
```

Comment:
I'm not sure on our stance on how much input validation to do, but if the malformed XML doesn't have the "gateway-receiver" element then I think we would get an NPE on the next line when we call receiverNodes.getLength().  Maybe its reasonable to expect a customer wouldn't accidentally remove this element though.

Reply:
Thanks. Let me add a test for that case. Its certainly possible that there are no receivers in the xml.

Reply:
It works ok with no receivers in the xml:

Initial xml content:
```
<cache>
</cache>
```
Processed xml content:
```
<cache>
</cache>
```